base/shell.jl

```diff
@@ -255,60 +255,91 @@ shell_escape_posixly(args::AbstractString...) =
     sprint(print_shell_escaped_posixly, args...)
 
 
-function print_shell_escaped_winsomely(io::IO, args::AbstractString...)
-    first = true
-    for arg in args
-        first || write(io, ' ')
-        first = false
-        # Quote any arg that contains a whitespace (' ' or '\t') or a double quote mark '"'.
-        # It's also valid to quote an arg with just a whitespace,
-        # but the following may be 'safer', and both implementations are valid anyways.
-        quotes = any(c -> c in (' ', '\t', '"'), arg) || isempty(arg)
-        quotes && write(io, '"')
-        backslashes = 0
-        for c in arg
-            if c == '\\'
-                backslashes += 1
-            else
-                # escape all backslashes and the following double quote
-                c == '"' && (backslashes = backslashes * 2 + 1)
-                for j = 1:backslashes
-                    # backslashes aren't special here
-                    write(io, '\\')
-                end
-                backslashes = 0
-                write(io, c)
+"""
+    shell_escape_wincmd(s::AbstractString)
+    shell_escape_wincmd(io, s::AbstractString)
+
+The unexported `shell_escape_wincmd` function escapes the meta
+characters `()!^<>&|` processed by the Windows `cmd.exe` shell: it
+places a `^` in front of any metacharacter that follows an even number
+of quotation marks on the command line.
+
```

Comment:
```suggestion
The result is safe to pass as an argument to a command call being processed by `CMD.exe /c` and will be received verbatim by the target application (or this function will fail with an ArgumentError).
This function may be useful in concert with the `windows_verbatim` flag to [`Cmd`](@ref) when constructing process pipelines.

!warning
    The argument parsing done by CMD when calling batch files (either inside `.bat` files or as arguments to them) is not compatible with the output of this function.

!important
    Due to a peculiar behavior of the CMD parser/interpreter, each command after a literal `|` character
    (indicating a command pipeline) must have `shell_escape_wincmd` applied twice since it will be parsed twice by CMD. For example:
    ```
    to_print = "All for 1 & 1 for all!"
    run(Cmd(Cmd(["cmd /c \"break | echo \$(shell_escape_wincmd(shell_escape_wincmd(to_print)))"]), windows_verbatim=true))
    ```
```
(adopting some of the advisory text from #33474):https://github.com/JuliaLang/julia/blob/09c1d61ae2ff77f53c0362a2a3cc31a2d5d95e3c/base/shell.jl#L324-L345

Reply:
I'm afraid I don't quite understand yet that suggested addition and I am not really familiar yet with all the behind-the-scenes processing that may go on during `run(Cmd(Cmd(["cmd /c \"...` on a Windows machine. My own use case and expertise is preparing command lines that get sent via ssh to a Windows server running OpenSSH sshd (e.g., in Distributed), where no libuv processing of command-line arguments passed to Windows processes is involved. I couldn't reproduce yet myself the effect referred to in this suggestion, and therefore would be more comfortable if we left this to a separate PR.

Reply:
> all the behind-the-scenes processing

Reply:
> all the behind-the-scenes processing

There is none. That's the point of using that syntax and flag: to disable all helpers and get the raw string without libuv processing or such.

Though I would perhaps further clarify here that it may be best passed as `CMD.exe /S /C " prog args "` (with the '/S' and surrounding " pair).